File: app/src/main/java/com/huanli233/biliterminal2/activity/live/FollowLiveActivity.java
package com.huanli233.biliterminal2.activity.live;

import android.annotation.SuppressLint;
import android.os.Bundle;

import com.huanli233.biliterminal2.activity.base.RefreshListActivity;
import com.huanli233.biliterminal2.adapter.LiveCardAdapter;
import com.huanli233.biliterminal2.api.LiveApi;
import com.huanli233.biliterminal2.bean.LiveRoom;
import com.huanli233.biliterminal2.util.ThreadManager;

import java.util.ArrayList;
import java.util.List;

public class FollowLiveActivity extends RefreshListActivity {
    private List<LiveRoom> roomList;
    private LiveCardAdapter adapter;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setPageName("我关注的直播");

        recyclerView.setHasFixedSize(true);

        roomList = new ArrayList<>();

        ThreadManager.run(() -> {
            try {
                roomList = LiveApi.getFollowed(page);
                adapter = new LiveCardAdapter(this, roomList);
                setOnLoadMoreListener(this::continueLoading);
                setRefreshing(false);
                setAdapter(adapter);
                if (roomList.isEmpty()) showEmptyView();
            } catch (Exception e) {
                loadFail(e);
            }
        });
    }

    private void continueLoading(int page) {
        ThreadManager.run(() -> {
            try {
                List<LiveRoom> list;
                list = LiveApi.getFollowed(page);
                runOnUiThread(() -> {
                    if (list != null) {
                        roomList.addAll(list);
                        adapter.notifyItemRangeInserted(roomList.size() - list.size(), list.size());
                    }
                });
                if (list != null && list.isEmpty()) {
                    setBottom(true);
                }
                setRefreshing(false);
            } catch (Exception e) {
                loadFail(e);
            }
        });
    }
}

File: app/src/main/java/com/huanli233/biliterminal2/activity/live/FollowLiveActivity.kt
package com.huanli233.biliterminal2.activity.live

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import com.huanli233.biliterminal2.activity.base.RefreshListActivity
import com.huanli233.biliterminal2.adapter.LiveCardAdapter
import com.huanli233.biliterminal2.api.LiveApi
import com.huanli233.biliterminal2.bean.LiveRoom
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class FollowLiveActivity : RefreshListActivity() {
    private val roomList = mutableListOf<LiveRoom>()
    private lateinit var adapter: LiveCardAdapter

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setPageName("我关注的直播")
        recyclerView.setHasFixedSize(true)

        loadInitialData()
    }

    private fun loadInitialData() {
        lifecycleScope.launch {
            try {
                val result = withContext(Dispatchers.IO) { 
                    LiveApi.getFollowed(page) 
                }
                
                roomList.clear()
                roomList.addAll(result)

                adapter = LiveCardAdapter(this@FollowLiveActivity, roomList)
                setAdapter(adapter)
                
                if (roomList.isEmpty()) {
                    showEmptyView()
                } else {
                    hideEmptyView()
                }
                
                setOnLoadMoreListener { page -> loadMoreData(page) }
                setRefreshing(false)
            } catch (e: Exception) {
                loadFail(e)
            }
        }
    }

    private fun loadMoreData(page: Int) {
        lifecycleScope.launch {
            try {
                val list = withContext(Dispatchers.IO) { 
                    LiveApi.getFollowed(page) 
                }
                
                if (!list.isNullOrEmpty()) {
                    val startPos = roomList.size
                    roomList.addAll(list)
                    adapter.notifyItemRangeInserted(startPos, list.size)
                }
                
                setRefreshing(false)
                setBottom(list.isNullOrEmpty())
            } catch (e: Exception) {
                loadFail(e)
            }
        }
    }
}
